src/main/kotlin/ecommerce/exception/GlobalExceptionHandler.kt

```diff
@@ -0,0 +1,66 @@
+package ecommerce.exception
+
+import ecommerce.exception.auth.EmailAlreadyExistsException
+import ecommerce.exception.auth.InvalidCredentialsException
+import ecommerce.exception.cartItem.CartItemNotFoundException
+import ecommerce.exception.product.DuplicateProductNameException
+import ecommerce.exception.product.ProductNotFoundException
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.security.authentication.BadCredentialsException
+import org.springframework.web.bind.MethodArgumentNotValidException
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestControllerAdvice
+
+@RestControllerAdvice
+class GlobalExceptionHandler {
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(MethodArgumentNotValidException::class)
+    fun handleValidationExceptions(ex: MethodArgumentNotValidException): ResponseEntity<Map<String?, Any>> {
+        val errors = ex.bindingResult.fieldErrors.associate { it.field to (it.defaultMessage ?: "Invalid Value") }
+        return ResponseEntity.badRequest().body(mapOf("errors" to errors))
+    }
+
+    @ExceptionHandler(ProductNotFoundException::class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    fun handleProductNotFound(ex: ProductNotFoundException): Map<String, String> {
+        return mapOf("error" to ex.message.orEmpty())
+    }
+
+    @ExceptionHandler(DuplicateProductNameException::class)
+    @ResponseStatus(HttpStatus.CONFLICT)
+    fun handleDuplicateException(ex: DuplicateProductNameException): Map<String, String> {
+        return mapOf("error" to ex.message.orEmpty())
+    }
```

Comment:
```suggestion
    @ExceptionHandler(value = [
        DuplicateProductNameException::class,
        EmailAlreadyExistsException::class,
    ])
    @ResponseStatus(HttpStatus.CONFLICT)
    fun handleConflictException(ex: RuntimeException): Map<String, String> {
        return mapOf("error" to ex.message.orEmpty())
    }
```

You can group these exceptions together like this, as they return the same HTTP status.

Reply:
Good point—grouping them does make the code cleaner and avoids repetition. I'll update it accordingly, thank you.